src/algebra/pointwise.lean

```diff
@@ -249,6 +249,32 @@ begin
   exact mul_le_mul' (hbA hxa) (hbB hxb),
 end
 
+open_locale big_operators
+
+/-- The n-ary version of `set.mem_mul`. -/
+@[to_additive /-" The n-ary version of `set.mem_add`. "-/]
+lemma mem_finset_prod {α ι : Type*} (t : finset ι) (f : ι → set α) [comm_monoid α] (a : α) :
```

Comment:
Wouldn't `mem_prod` be the expected name here? Or is that already taken?

Would it make sense to also have `prod_mem_prod` (where `a` is of the form `∏ i in t, g i` already.

Reply:
I didn't want to use `set.mem_prod` because I would expect that to be about `set.prod`. I'll try to add the `prod_mem_prod` lemma

Reply:
`finset_prod_mem_finset_prod` was easy, as was `finset_prod_subset_finset_prod ` - mind taking another look?